app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -60,10 +55,14 @@ open class FenixApplication : Application() {
             // If this is not the main process then do not continue with the initialization here. Everything that
             // follows only needs to be done in our app's main process and should not be done in other processes like
             // a GeckoView child process or the crash handling process. Most importantly we never want to end up in a
-            // situation where we create a GeckoRuntime from the Gecko child process (
+            // situation where we create a GeckoRuntime from the Gecko child process.
             return
         }
 
+        // We want to call this function as early as possible, but only once and
+        // on the main process, as it uses Gecko to fetch experiments from the server.
+        experimentLoader = loadExperiments()
```

Comment:
+1 Let's only initialize this in the main process.

We probably want to do the same for `setDayNightTheme()` (No matter what it should happen after the crash reporter gets initialized otherwise crashes from this will go into the void). But that's unrelated and for a follow-up.